compiler/ilgen/TypeDictionary.cpp

```diff
@@ -421,6 +431,15 @@ TypeDictionary::GetFieldType(const char *structName, const char *fieldName)
    return theStruct->getFieldType(fieldName);
    }
 
+uint32_t
+TypeDictionary::OffsetOf(const char *structName, const char *fieldName)
+   {
+   TR_HashId structID = 0;
+   _structsByName->locate(structName, structID);
```

Comment:
what behaviour should this function have if a valid structName or fieldName isn't provided? I know the rest of the file doesn't always handle this case, but we should start thinking about it as we augment the API. Perhaps for now it can return an offset 0 (as clearly a "wrong" result but at least it keeps any subsequent memory access inside the given object unlike what, say, UINT_MAX would).

Or is the intention to deal with all these problems together in a separate commit?